geopandas/plotting.py

```diff
@@ -283,8 +283,8 @@ def _plot_point_collection(
     geoms, multiindex = _flatten_multi_geoms(geoms)
     # values are expanded below as kwargs["c"]
 
-    x = [p.x for p in geoms]
-    y = [p.y for p in geoms]
+    x = [p.x if not p.is_empty else None for p in geoms]
+    y = [p.y if not p.is_empty else None for p in geoms]
```

Comment:
Do we know here we already can't have None?

Reply:
Ah, yes, because at a higher level we explicitly selected the points

Reply:
This only covers empty point cases, so we have Point object but it has no coordinates to return and errors.